www/android/geolocation.js

```diff
@@ -38,13 +39,14 @@ module.exports = {
                 error(new PositionError(PositionError.PERMISSION_DENIED, 'Illegal Access'));
             }
         };
-        exec(win, fail, 'Geolocation', 'getPermission', []);
+        exec(win, fail, 'Geolocation', 'getPermission', [args.enableHighAccuracy]);
     },
 
     watchPosition: function (success, error, args) {
         var pluginWatchId = utils.createUUID();
 
-        var win = function () {
+        var win = function (deviceApiLevel) {
+            if (deviceApiLevel === 31) args.enableHighAccuracy = true; // Workaround for bug specific to API 31 where requesting `enableHighAccuracy: false` results in TIMEOUT error.
```

Comment:
here `args` is optional and may be undefined. Therefore we need to check `args` first.